servemux.go

```diff
@@ -16,18 +16,28 @@ type ServeMux struct {
 // ctxKey is the key type used for path parameters in the request context
 type ctxKey string
 
-// ctxRouteNameKey is the type used for the route in the request context.
-// Distinct from ctxKey to avoid collisions
-type ctxRouteNameKey string
-
-var routeKey = ctxRouteNameKey("route_name")
+var (
+	routeKey = ctxKey("route_name")
+	paramKey = ctxKey("params")
+)
 
 // PathParam gets named path parameters and their values from the request
 //
 // the path '/users/:name' given '/users/andrew' will have `PathParam(r, "name")` => `"andrew"`
 // unset values return an empty stringRoutes
 func PathParam(req *http.Request, name string) (value string) {
-	value, _ = req.Context().Value(ctxKey(name)).(string)
+	value = req.Context().Value(paramKey).(map[string]string)[name]
```

Comment:
You can create a private method like 
```
def pathParams(req *http.Request) map[string]string {
  return req.Context().Value(paramKey).(map[string]string)
}
```
and use it in both `PathParam` and `PathParams`.